source/gui/blockAction.py

```diff
@@ -0,0 +1,58 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import config
+from dataclasses import dataclass
+from enum import Enum
+from functools import wraps
+import globalVars
+from typing import Callable
+import ui
+from gui.message import isModalMessageBoxActive
+
+
+@dataclass
+class _Context:
+	blockActionIf: Callable[[], bool]
+	translatedMessage: str
+
+
+class Context(_Context, Enum):
+	SECURE_MODE = (
+		lambda: globalVars.appArgs.secure,
+		# Translators: Reported when an action cannot be performed because NVDA is in a secure screen
+		_("Action unavailable in secure context"),
+	)
+	WINDOWS_STORE_VERSION = (
+		lambda: config.isAppX,
+		# Translators: Reported when an action cannot be performed because NVDA has been installed
+		# from the Windows Store.
+		_("Action unavailable in NVDA Windows Store version"),
+	)
+	MODAL_DIALOG_OPEN = (
+		isModalMessageBoxActive,
+		# Translators: Reported when an action cannot be performed because NVDA is waiting
+		# for a response from a modal dialog
+		_("Action unavailable as an open dialog requires a response"),
+	)
+
+
+def when(*contexts: Context):
+	"""Returns a function wrapper.
+	A function decorated with `when` will exit early if any supplied context in `contexts` is active.
+
+	For example, a function decorated with `@blockAction.when(blockAction.Context.SECURE_MODE)`
+	will return without running if secure mode is active.

```

Comment:
Please add an example and highlight that the order is important here, only the first to block will be reported. This means that more permanent conditions should be specified before temporary ones. EG appx before dialog open.

Reply:
I've squashed and rebased the approved commit onto master in `b5e171b0d8f42542bd1edf863cc2b74330fc8c85` can you approve the wording and ordering change in 2668b6623d99e00a841e46046ea3bf0185d6d027?